corehq/apps/reports/management/commands/create_tableau_connection.py

```diff
@@ -0,0 +1,220 @@
+            self.sign_out()
+        # I think it's ok to ignore the flake8 error for this bare except clause since we re-raise the exception.
+        except:  # noqa: E722
+            logger.info("Command execution interrupted.")
+            # Submit a sign out request on any interruption to ensure the token we were given upon sign in is no
+            # longer valid.
+            self.sign_out()
+            raise
+
+    def sign_in(self):
+        logger.info("Signing into the API so we can add a new connected app...")
+        request_url = f"{self.url_base}/auth/signin"
+        payload = {
+            "credentials": {
+                "personalAccessTokenName": self.options['pat_name'],
+                "personalAccessTokenSecret": self.options['pat_secret'],
+                "site": {
+                    "contentUrl": self.site_name
+                }
+            }
+        }
+        response_body = self._process_response(
+            response=requests.post(request_url, data=json.dumps(payload), headers=self.headers),
+            custom_msg_if_failure="Request to sign in failed."
+        )
+        api_auth_token = response_body["credentials"]["token"]
+        self.headers.update({"X-Tableau-Auth": api_auth_token})
+        logger.info("Sign in successful.")
+
+    def sign_out(self):
+        logger.info('Signing out of the API...')
+        try:
+            self.request_sign_out()
+            logger.info('Sign out request successful.')
+        except RequestFailure as e:
+            logger.info(e)
+
+    def request_sign_out(self):
+        request_url = f"{self.url_base}/auth/signout"
+        self._process_response(
+            response=requests.post(request_url, headers=self.headers),
+            custom_msg_if_failure="Request to sign out failed.",
+            returns_body=False
+        )
+
+    def get_site_id(self):
+        request_url = f"{self.url_base}/sites/{self.site_name}?key=name"
+        response_body = self._process_response(
+            response=requests.get(request_url, headers=self.headers),
+            custom_msg_if_failure="Request to get site ID from name failed."
+        )
+        return response_body["site"]["id"]
+
+    def check_if_connected_app_exists(self):
+        # Use the list apps api method to check if a connected app exists with the name 'commcarehq_embedding'.
+        request_url = f"{self.url_base}/sites/{self.site_id}/connected-applications"
+        response_body = self._process_response(
+            response=requests.get(request_url, headers=self.headers),
+            custom_msg_if_failure="Request to check if a connected app already exists failed."
+        )
+        if response_body["connectedApplications"]:
+            connected_apps = response_body["connectedApplications"]["connectedApplication"]
+            for app in connected_apps:
+                if app["name"] == "commcarehq_embedding":
+                    return (True, app["clientId"])
```

Comment:
nit: tuple braces are unnecessary
```suggestion
                    return True, app["clientId"]
```